fsd_utils/authentication/decorators.py

```diff
@@ -49,6 +51,18 @@ def _check_access_token(auto_redirect=True):
         return False
 
 
+def _set_user(token_payload):
+    full_name = token_payload.get("fullName")
+    email = token_payload.get("email")
+    roles = token_payload.get("roles")
+    return User(
+        full_name=full_name,
+        email=email,
+        roles=roles,
+        highest_role=get_highest_role(roles),
+    )
+
+
 def login_required(f):
```

Comment:
just a QQ of curiosity: is the purpose of login_required/requested to supply the method it wraps with the flask g global variable and all the user authentication properties it carries?

Reply:
Yep. The flask g variable is always there but these decorators just add the user properties to it.